source/slang/slang-check-inheritance.cpp

```diff
@@ -0,0 +1,945 @@
+// slang-check-inheritance.cpp
+#include "slang-check-impl.h"
+
+// This file implements the semantic checking logic
+// related to computing linearized inheritance
+// information for types and decalrations.
+
+//#include "slang-lookup.h"
+//#include "slang-syntax.h"
+//#include "slang-ast-synthesis.h"
+//#include <limits>
+
+namespace Slang
+{
+    InheritanceInfo SharedSemanticsContext::getInheritanceInfo(Type* type)
```

Comment:
Does it make sense to store the InheritanceInfo directly inside a `Type` object?

Reply:
There's a really subtle thing here, that the `InheritanceInfo` for a type does *not* depend only on the `Type`.

We create a distinct `SharedSemanticsContext` for each module that we check, and one of the reasons for that is that the set of `import`ed modules has an effect on what `extension`s are visible during checking. When different sets of `extension`s are visible, it is possible for the same `Type` to yield different linearizations (which makes perfect sense; it has different (transitive) bases and members).